### ISSUE TITLE
Add-Type Assembly - fails due to OOP [classes]

### ISSUE THREAD

**Issue by SibTiger** [SibTiger/PowerShell-Compact-Archive-Tool#34]
Upon loading, the parser will fail at CommonGUI due to unknown datatypes.  This is because - the Add-Types had not yet been executed due to POSH's parsing mechanism.

Source:
https://stackoverflow.com/questions/42837447/powershell-unable-to-find-type-when-using-ps-5-classes

**Comment by SibTiger** [comment]
Two ways I can approach this:
1. Use a two part startup: add-type(s) then load PSCAT.
2. CONVERT commonGUI from OOP to POP.


Then after figuring this out, kick myself for using POSH.

**Comment by SibTiger** [comment]
This is now closed.

I decided to take the first option from this [post](https://github.com/SibTiger/PowerShell-Compact-Archive-Tool/issues/34#issuecomment-1100606674).  Instead of starting the PSCAT directly, the user will now have to start the Launcher.  This was a necessary step to assure that the Powershell's environment is prepared for the PowerShell Compact-Archive Tool application; now I can add as many dependencies as I wish!!!  I AM UNSTOPABLE!!!

Startup Sequence Map;
------------------------------

1. User Clicks on the Launcher.ps1
2. Launcher Starts
3. Launcher prepares the environment using `Add-Type -Assembly` and the like.
4. Launcher Starts PSCAT with appropriate arguments (optional from the user)
5. PSCAT runs normally
6. {{USER DOES STUFF WITH PSCAT}}
7. PSCAT terminates
8. Launcher terminates
